src/main/java/chatbot/TaskList.java

```diff
@@ -0,0 +1,99 @@
+package chatbot;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * The TaskList class manages a list of tasks. It provides methods to manipulate tasks
+ * such as adding, removing, and marking them as done or not done. It also allows accessing
+ * individual tasks and the entire list of tasks.
+ */
+
+public class TaskList {
+
+  protected ArrayList<Task> al;
+
+  /**
+   * Constructs a TaskList with the specified list of tasks.
+   *
+   * @param al The initial list of tasks.
+   */
+  public TaskList(ArrayList<Task> al) {
+    this.al = al;
+  }
+
+  /**
+   * Adds a task to the task list.
+   *
+   * @param t The task to be added.
+   */
+  public void add(Task t) {
```

Comment:
Perhaps you can rename the variable `t` to something more descriptive? `task` should be fine